### PR TITLE
Replace deprecated Google OAuth OOB flow

### LINKED ISSUE #12455: Google add-ons use deprecated OAuth out-of-band (flow)
Some add-ons using Google APIs still use the deprecated OOB flow. It is no longer possible to create new Google OAuth tokens using the OOB flow since February 28th.

See: https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#disallowed-oob

As a result it is no longer possible to setup the following add-ons:

* Google Cloud Text-to-Speech
* Google Cloud Speech-To-Text
* Nest Binding

These need to be changed to use one of the other flows.

### PR DESCRIPTION
**Pull request by wborn**
This replaces the deprecated OOB OAuth flow used in the Nest Binding and Google STT/TTS add-ons.

Instead they will now use https://www.google.com as redirect_uri.
The instructions in the README and config descriptions have been updated for this change.

See: https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#disallowed-oob

Fixes #12455

### CODE REVIEW

**Review by GiviMAD — APPROVED** [review]
Thank you!!

**Review by kaikreuzer — APPROVED** [review]
Thanks!

File: bundles/org.openhab.binding.nest/README.md [hunk 1]
```diff
@@ -67,8 +67,9 @@ Now an OAuth 2.0 client is created and configured for using the SDM API by the b
 1. Create a new client on the "Credentials" page ([APIs & Services > Credentials](https://console.cloud.google.com/apis/credentials)):
     1. Click the "Create Credentials" button at the top of the page
     1. Choose "OAuth client ID"
-    1. As "Application type" choose "TVs and Limited Input devices"
+    1. As "Application type" choose "Web application"
     1. Give it a name so you can remember what it is used for (e.g. `Nest Binding SDM`)
+    1. Add "https://www.google.com" to the "Authorized redirect URIs"
     1. Click "Create" to create the client
     1. Copy and save the generated **Client ID** (e.g. `1046297811237-3f5sj4ccfubit0fum027ral82jgffsd1.apps.googleusercontent.com`) and **Client Secret** (e.g. `726kcU-d1W4RXxEJA79oZ0oG`) somewhere
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.nest/README.md:74]: @wborn Can you confirm that these credentials are fake? If so, could we change them to be obviously fake ones?

Comment (wborn) [reply bundles/org.openhab.binding.nest/README.md:74]: Yes they are fake. They resemble the real thing so it is clear what to enter. Does it trigger some kind of security scan?

Comment (kaikreuzer) [reply bundles/org.openhab.binding.nest/README.md:74]: Yeah, exactly - it was an enterprise-internal vulnerability scan; apparently purely triggered by the fact that I have write access to this repo here and also use the same GitHub handle for professional contributions on other repos.

**Pull request merged**